ble/services/EddystoneService.h

```diff
@@ -185,6 +185,29 @@ class EddystoneService
         return false;
     }
 
+    /**
+     *  Set Eddystone URL Frame information.
+     *  @param[in] power          TX Power in dB measured at 0 meters from the device.
+     *  @param[in] url            Encoded URL
+     *  @param[in] urlAdvPeriodIn How long to advertise the URL frame (measured in # of adv periods)
+     *  @return false on success, true on failure.
+     */
+    bool setURLFrameEncodedData(int8_t power, const char *encodedUrlIn, uint8_t encodedUriInLength, uint32_t urlAdvPeriodIn) {
+        if (0 == urlAdvPeriodIn) {
+            urlIsSet = false;
+            return false;
+        }
+        defaultUrlPower = power;
+        memcpy(defaultUriData, encodedUrlIn, URI_DATA_MAX);
```

Comment:
I think this would introduce a bug. Recall that the encoding for ".com/" is 0x0, which is equivalent to "\0" then with the encoding of a URL like "http://mbed.com/hello" the last 5 characters will be dropped. This is because according to the documentation for strncpy():

`If the end of the source C string (which is signaled by a null-character) is found before num characters have been copied, destination is padded with zeros until a total of num characters have been written to it.`